resources/2023-11-application-diagnostics-prototype.md

```diff
@@ -0,0 +1,264 @@
+
+The diagnostics store configuration is out-of-band of Radius. Since we're building a demo it's helpful to describe what it will do.
+
+This includes:
+
+- Configuring Azure Monitor as a store of diagnostics.
+- Configuring Grafana as a dashboard, using Azure Monitor as a data source.
+  - Creating alerts and visualizations for our demo.
+- Installing and configuring the OpenTelemetry collector Helm Chart.
+  - Understanding Radius' labels and enriching data.
+  - Sending data to the Azure Monitor store.
+
+### API design (if applicable)
+
+Here's a speculative design of the API for new settings. This will be improved as we iterate.
+
+To briefly cover what settings are defined here (net new):
+
+- Enabling/Disabling the OpenTelemetry collector, and saying how it's managed (eg: `external`, `none`).
+- Providing the destination address for the OpenTelemetry collector inside the cluster. 
+- Describing for documentation purposes where to find diagnostics data. This is to support tooling experiences.
+- Describing a range of diagnostics store technologies for Use in Recipes. This example includes Azure Monitor.
+
+```tsp
+@doc("Environment properties")
+model EnvironmentProperties {
+  @doc("The status of the asynchronous operation.")
+  @visibility("read")
+  provisioningState?: ProvisioningState;
+
+  @doc("The compute resource used by application environment.")
+  compute: EnvironmentCompute;
+
+  @doc("Cloud providers configuration for the environment.")
+  providers?: Providers;
+
+  @doc("Simulated environment.")
+  simulated?: boolean;
+
+  @doc("Specifies Recipes linked to the Environment.")
+  recipes?: Record<Record<RecipeProperties>>;
+
+  @doc("The environment extension.")
+  @extension("x-ms-identifiers", [])
+  extensions?: Array<Extension>;
+ 
+  // NEW PROPERTY
+  @doc("Specifies settings for diagnostics collection")
+  diagnostics?: EnvironmentDiagnostics;
+}
+
+// NEW TYPE
+model EnvironmentDiagnostics {
+  @doc("Configures settings related to OpenTelemetry")
+  openTelemetry?:  EnvironmentOpenTelemetryDiagnostics;
+
+  @doc("Configures the location where diagnostics for the environment can be viewed")
+  primaryDashboard: EnvironmentDashboardDiagnostics;
+
+  @doc("Configures settings related to Azure monitor")
+  azureMonitor?: EnvironmentAzureMonitorDiagnostics;
+}
+
+// NEW TYPE
+model EnvironmentOpenTelemetryDiagnostics {
+  // External means the collector is present, but was configured out-of-band with Radius.
+  @doc("Specifies the mode of the OpenTelemetry collector")
+  collectorMode: 'none' | 'external';
+
+  // Administrator sets this so it gets injected into the pods.
+  @doc("Specifies the address of the OpenTelemetry collector")
+  collectorAddress: string;
+}
+
+// NEW TYPE
+model EnvironmentDashboard {
+  @doc("URL of the dashboard where diagnostics can be viewed.")
+  url: string
+}
+
+// NEW TYPE
+model EnvironmentAzureMonitorDiagnostics  {
+  @doc("The resource ID of the Azure Monitor instance")
+  id: string
```

Comment:
Could you explain how the resource ID will be used?

Reply:
For now I was planning to use this for informational purposes. For example we could *deep-link* from the portal. This is part of my goal of "making the implicit explicit". 

Open to suggestions if you have other ideas 👍